lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -29,6 +32,144 @@
  * of patches to that input. Tokenizes HTML but does not fully
  * parse the input document.
  *
+ * ## Usage
+ *
+ * Use of this class requires three steps:
+ *
+ *  1. Create a new class instance with your input HTML document.
+ *  2. Find the tag(s) you are looking for.
+ *  3. Request changes to the attributes in those tag(s).
+ *
+ * Example
+ * ```php
+ *     $processor = new WP_HTML_Tag_Processor( $html );
+ *     if ( $processor->next_tag( array( 'tag_name' => 'option' ) ) ) {
```

Comment:
What do you think about surfacing the shorthand form?

```suggestion
 *     if ( $processor->next_tag( 'option' ) ) {
```

Reply:
definitely not in this PR, but overall I'm not sure.

on one hand I think it's convenient, but on the other it does muddy the consistency, and I think it's likely more common that people will search by class anyway rather than tag, or at least reasonably as often as by class.

it did occur to me that maybe `tag` and `class` would suffice, instead of `tag_name` and `class_name`

on this note I wonder if we have to stick to the long array syntax in our example code. we have to use those in the code for PHP support, but are we required to demonstrate as examples the use of the more restricted syntax?

Reply:
> I think it's likely more common that people will search by class anyway rather than tag, or at least reasonably as often as by class.

I'd say 50/50 is as good of a guess as we can make right now.

> it did occur to me that maybe tag and class would suffice, instead of tag_name and class_name

Yes please!

>  are we required to demonstrate as examples the use of the more restricted syntax?

That is a good question, I don't know! @costdev what do you think?

Reply:
There's ~11 examples of short array syntax within docblocks/inline comments, and ~53 examples of long array syntax. While [this example](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#description) (see point 2) shows the long syntax, the accompanying text doesn't have anything to do with the "right" syntax to use. I'd suggest just going with best judgement on readability.

However, pinging @SergeyBiryukov as someone who has probably seen more Core docblocks than anyone.

Reply:
I've replace the examples with shorthand array syntax for brevity and clarity. We can remove them later if we decide that it's best to leave in the `array()` notation.

31402ddfc8